stdlib/builtin.rbi

```diff
@@ -560,10 +560,60 @@ class Enumerator<'a, 'b>
                 | -> Enumerator<['a, Integer], 'b>
 end
 
+class Encoding
+end
+
 class Regexp
   def self.compile: (String, *any) -> Regexp
+                  | (Regexp) -> Regexp
   def self.escape: (String) -> String
+  def self.last_match: -> MatchData?
+                     | (Integer) -> String?
+  def self.quote: (String) -> String
+  def self.try_convert: (any) -> Regexp?
+  def self.union: (*String) -> Regexp
+                | (Array<String>) -> Regexp
+                | (*Regexp) -> Regexp
+                | (Array<Regexp>) -> Regexp
+
+  def initialize: (String, *any) -> any
+                  | (Regexp) -> any
+  def ===: (String) -> bool
+  def =~: (String) -> Integer?
+  def casefold?: -> bool
+  def encoding: -> Encoding
+  def fixed_encoding?: -> bool
+  def match: (String) -> MatchData?
+           | (String, Integer) -> MatchData?
+           | (String) { (MatchData) -> void } -> void
+           | (String, Integer) { (MatchData) -> void } -> void
+  def match?: (String) -> bool
+            | (String, Integer) -> bool
+  def named_captures: -> Hash<String, Array<Integer>>
+  def names: -> Array<String>
+  def options: -> Integer
   def source: -> String
+  def ~: -> Integer?
+end
+
+class MatchData
+  def []: (Integer | String | Symbol) -> String?
+        | (Integer, Integer) -> Array<String>
+        | (Range<Integer>) -> Array<String>
+  def begin: (Integer | String | Symbol) -> Integer
```

Comment:
`... -> Integer?`

Reply:
Oh, this may be a ducumentation's bug. 💦 

Documentation says `begin(n) → integer`, but correctly `begin(n) → integer or nil`.

http://ruby-doc.org/core-2.5.1/MatchData.html#method-i-begin

Anyways, I will fix the `begin` definition. Thanks!